libs/payments/ui/src/lib/client/components/PurchaseDetails/index.tsx

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use client';
+
+import { Localized } from '@fluent/react';
+import Image from 'next/image';
+import { useState } from 'react';
+import chevron from './images/chevron.svg';
+
+type PurchaseDetailsProps = {
+  children: React.ReactNode;
+  priceInterval: React.ReactNode;
+  purchaseDetails: {
+    details: string[];
+    subtitle: string | null;
+    productName: string;
+    webIcon: string;
+  };
+};
+
+export function PurchaseDetails(props: PurchaseDetailsProps) {
+  const { children, priceInterval, purchaseDetails } = props;
+  const { subtitle, productName, webIcon } = purchaseDetails;
+  const [detailsHidden, setDetailsState] = useState(false);
+  return (
+    <div className="bg-white rounded-b-lg shadow-sm shadow-grey-300 text-sm px-4 rounded-t-none tablet:rounded-t-lg">
+      <div className="flex gap-4 my-0 py-4 border-b border-grey-200">
+        <Image
+          src={webIcon}
+          alt={productName}
+          data-testid="product-logo"
+          className="w-16 h-16 rounded-lg"
+          width={64}
+          height={64}
+        />
+
+        <div className="text-start">
+          <h2 className="text-grey-600 font-semibold leading-5 my-0 break-words">
+            {productName}
+          </h2>
+
+          <p className="text-grey-400 mt-1 mb-0">
+            {priceInterval}
+            {subtitle && (
+              <span>
+                &nbsp;&bull;&nbsp;
+                {subtitle}
+              </span>
+            )}
+          </p>
+        </div>
+      </div>
+
+      <div className={detailsHidden ? 'hidden tablet:block' : 'block'}>
+        {children}
+      </div>
+
+      <div
+        className="flex items-center justify-center tablet:hidden"
+        data-testid="purchase-details-footer"
+      >
+        {detailsHidden ? (
+          <button
+            className="flex items-center justify-center bg-transparent border-transparent cursor-pointer text-blue-500 leading-5 my-2 py-2 px-2 relative focus:border focus:border-solid focus:border-blue-400 focus:py-2 focus:px-4 focus:rounded-md focus:shadow-none"
```

Comment:
nit. could there be a white (read: effectively invisible) border around the button when it's not focused? Currently after clicking the button, when you click out of the button on the page, the border disappears and there's a visual item shift.